lsp/server/src/test/TestHelper.ts

```
@@ -0,0 +1,153 @@
+ */
+
+import { Stats } from 'fs';
+import { mkdtemp, rm, stat } from 'fs/promises';
+import * as os from 'os';
+import * as path from 'path';
+import * as process from 'process';
+import * as sinon from 'sinon';
+import Account from '../../testFixture/objectInfos/Account.json';
+import {
+    ConfigAggregator,
+    StateAggregator,
+    Connection,
+    AuthInfo
+} from '@salesforce/core';
+import { WorkspaceUtils } from '../utils/workspaceUtils';
+
+const testOrgConfig = {
+    key: 'target-org',
+    value: 'testOrg'
+};
+
+const noOrgConfig = {
+    key: 'target-org'
+};
+
+export class TempProjectDirManager {
+    readonly projectDir: string;
+
+    private constructor(projectDir: string) {
+        this.projectDir = projectDir;
+    }
+
+    async removeDir(): Promise<void> {
+        return new Promise(async (resolve, reject) => {
+            let projectDirStats: Stats;
+            try {
+                projectDirStats = await stat(this.projectDir);
+            } catch (err) {
+                return reject(
+                    `Project dir '${this.projectDir}' does not exist or is inaccessible.`
+                );
+            }
+            if (!projectDirStats.isDirectory()) {
+                return reject(
+                    `Project dir '${this.projectDir}' is not a directory.`
+                );
+            }
+            await rm(this.projectDir, { recursive: true, force: true });
+            return resolve();
+        });
+    }
+
+    static async createTempProjectDir(): Promise<TempProjectDirManager> {
+        return new Promise(async (resolve, reject) => {
+            try {
+                const projectDir = await mkdtemp(
+                    path.join(os.tmpdir(), 'offlineWizard-')
+                );
+                return resolve(new TempProjectDirManager(projectDir));
+            } catch (err) {
+                return reject(err);
+            }
+        });
+    }
```

```suggestion
        const projectDir = await mkdtemp(path.join(os.tmpdir(), 'offlineWizard-'));
        return new TempProjectDirManager(projectDir);
```

---

Return a value from a function that returns promise should automatically wrap the return value.

For error, you're re-throwing the error in the form `return reject(err)` anyway so might as well just run the code and let it throw on its own and let the caller just handle it.